### PR TITLE
Pybind11

### PR DESCRIPTION
**Pull request by sanderboer**

### CODE REVIEW

File: srcpkgs/python3-pybind11/template [hunk 1]
```diff
@@ -23,5 +21,5 @@ checksum=1eed57bc6863190e35637290f97a20c81cfe4d9090ac0a24f3bbf08f265eb71d
 
 post_install(){
 vlicense LICENSE LICENSE
-
   }
+
```
Comment (misuchiru03) [inline srcpkgs/python3-pybind11/template]: Looks like you accidentally added an empty line at the end of the file.

Comment (misuchiru03) [inline srcpkgs/python3-pybind11/template]: Looks like you accidentally added an empty line at the end of the template.

Comment (misuchiru03) [reply srcpkgs/python3-pybind11/template]: And the license should be [BSD-3-Clause](https://opensource.org/licenses/BSD-3-Clause)

Comment (sanderboer) [reply srcpkgs/python3-pybind11/template]: Fixed these, thank you.

**Comment by NeelChotai** [comment]
I don't think this should require the `python3` prefix, you could remove that and the `wrksrc` line.

`short_desc` should read `lightweight header-only library that exposes C++ types in Python and vice versa`

Cross-compilation may be related to https://github.com/pybind/pybind11/issues/1330 and https://github.com/pybind/pybind11/issues/99.

File: srcpkgs/python3-pybind11/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+depends="boost python3"
+short_desc="A lightweight header-only library that exposes C++ types in Python and vice versa."
+maintainer="Sander Boer <sanderboer@mauc.nl>"
+license="BSD-3-Clause"
+homepage="https://github.com/pybind/pybind11"
+distfiles="https://github.com/pybind/pybind11/archive/v${version}.tar.gz"
+checksum=1eed57bc6863190e35637290f97a20c81cfe4d9090ac0a24f3bbf08f265eb71d
+
+post_install(){
+vlicense LICENSE LICENSE
```
Comment (yopito) [inline srcpkgs/python3-pybind11/template]: `vlicense LICENSE` is sufficient

**Comment by yopito** [comment]
I have a WIP that is crossbuilding with these:
```
build_style=cmake
configure_args="-DPYTHON_EXECUTABLE=/usr/bin/python3"
hostmakedepends="python3-devel python3-pytest"
makedepends="python3-devel boost-devel"
```

File: srcpkgs/python3-pybind11/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+-DPYTHON_LIBRARY=/usr/lib/libpython3.8.so
+-DUSE_PYTHON_INCLUDE_DIR=ON
+"
+hostmakedepends=""
+makedepends="python3-pytest eigen boost-devel python3-devel"
+depends="boost python3"
+short_desc="A lightweight header-only library that exposes C++ types in Python and vice versa."
+maintainer="Sander Boer <sanderboer@mauc.nl>"
+license="BSD-3-Clause"
+homepage="https://github.com/pybind/pybind11"
```
Comment (yopito) [inline srcpkgs/python3-pybind11/template]: `homepage="https://pybind11.readthedocs.io/en/latest"`, no ?

**Comment by gspe** [comment]
This template doesn't build python files and has unnecessary dependency `depends="boost python3" `

The follow template works, and should cross-compile:

```
# Template file for 'pybind11'
pkgname=pybind11
version=2.4.3
revision=1
build_style=cmake
configure_args="-D PYTHON_INCLUDE_DIR=ON -D PYBIND11_MASTER_PROJECT=ON -D PYBIND11_TEST=OFF"
hostmakedepends="python3-devel python3-setuptools"
makedepends="python3-devel"
short_desc="Seamless operability between C++11 and Python"
maintainer="Giuseppe Fierro <gspe@ae-design.ws>"
license="BSD-3-Clause"
homepage="https://pybind11.readthedocs.io"
distfiles="https://github.com/pybind/${pkgname}/archive/v${version}.tar.gz"
checksum=1eed57bc6863190e35637290f97a20c81cfe4d9090ac0a24f3bbf08f265eb71d

# build python package
post_build() {
	# cmake already install pybind11 headers
	export PYBIND11_USE_CMAKE=ON

	python3 setup.py build

	python3 setup.py install --prefix=/usr --root=${DESTDIR}
}

post_install() {
	vlicense LICENSE
}

```

**Comment by ahesford** [comment]
This is resolved by #20218; the CMake modules are fundamentally broken in `pybind11` and, until they are fixed upstream, are excluded from the Void project. [Upstream issue #2139](https://github.com/pybind/pybind11/issues/2139) tracks this problem.

**Pull request closed without merging**